### PR TITLE
common/vospi: Set vospi bus CPOL=1 and CPHA=1.

### PR DESCRIPTION
**Pull request by kwagyeman**
Fixes: https://github.com/openmv/openmv/issues/2485

1. Fixes swapped clock polarity names for the RT1062.
2.  The FLIR Lepton needs:

```
spi_config.clk_pol = OMV_SPI_CPOL_HIGH;
spi_config.clk_pha = OMV_SPI_CPHA_2EDGE;
```

image

On the STM32... we've been running for years with:

```
config->clk_pol = OMV_SPI_CPOL_LOW;
config->clk_pha = OMV_SPI_CPHA_1EDGE;
```

However, things work with the new settings, too... weirdly enough on the STM32.

...

Need to test VOSPI/FIR  thoroughly on both boards and all SPI IMXRT devices before this is ready.

### CODE REVIEW

**Comment by iabdalkader** [comment]
This pull request has been mentioned on **OpenMV Forums**. There might be relevant details there:

https://forums.openmv.io/t/noise-when-turning-off-automatic-gain-control-on-the-flir-lepton/10163/12

**Comment by kwagyeman** [comment]
Tested:
* FLIR Lepton 3.5 via Sensor on H7 Plus
* FLIR Lepton 1.0/3.5 via FIR on H7 Plus
* FLIR Lepton 3.5 via Sensor on RT1060
* FLIR Lepton 1.0/3.5 via FIR on RT1060
* LCD Shield on the RT1062
* TV Shield on the RT1062
* PAJ6100 on the RT1062

**Comment by kwagyeman** [comment]
Fixes: https://github.com/openmv/openmv/issues/2485

**Comment by iabdalkader** [comment]
> Fixes swapped clock polarity names for the RT1062.

The clocks are not swapped.

image
image

> On the STM32... we've been running for years with:

Like most things, it works with either SPI mode 1 or SPI mode 3.

Also FLIR works with RT1060, as the main sensor, using sensors/lepton and vospi. Whatever the problem is, it's in py_fir.

image

File: src/omv/ports/mimxrt/omv_spi.c [hunk 1]
```diff
@@ -313,7 +313,7 @@ int omv_spi_default_config(omv_spi_config_t *config, uint32_t bus_id) {
     config->spi_mode = OMV_SPI_MODE_MASTER;
     config->bus_mode = OMV_SPI_BUS_TX_RX;
     config->bit_order = OMV_SPI_MSB_FIRST;
-    config->clk_pol = OMV_SPI_CPOL_HIGH;
+    config->clk_pol = OMV_SPI_CPOL_LOW;
```
Comment (iabdalkader) [inline src/omv/ports/mimxrt/omv_spi.c:316]: The default should be SPI Mode 0.

**Review by iabdalkader — APPROVED** [review]

**Pull request merged**